wavelength-plugins/badges.js

```diff
@@ -1,134 +1,148 @@
+const TD_CSS = 'style ="background: rgba(69, 76, 80, 0.6); color: #FFF; padding: 5px; border: 1px solid #222; border: 3px solid #FFF; border-radius: 4px"';
 
 function badgeImg(IMG_URL, name) {
-	return '<img src="' + IMG_URL + '" height="16" width="16" alt="' + name + '" title="' + name + '" >';
+	return `<img src="${IMG_URL}" height="16" width="16" alt="${name}" title="${name}">`;
 }
 
 exports.commands = {
 	badge: 'badges',
-	badges: function (target, room, user) {
-		let parts = target.split(',');
-		let cmd = parts[0].trim().toLowerCase();
-		let targetUser;
-		let selectedBadge;
-		let userBadges;
-		let userid;
-		let output = '';
-		switch (cmd) {
-		case 'give':
-		case 'set':
+	badges: {
+		give: function (target, room, user) {
+			let targetUser;
+			let selectedBadge;
+			let userBadges;
+			let userid;
 			if (!this.can('lock')) return false;
-			if (parts.length !== 3) return this.errorReply("Correct command: `/badges set, [user], [badgeName]`");
-			userid = toId(parts[1].trim());
+			let parts = target.split(',');
+			if (parts.length !== 2) return this.parse("/badgeshelp");
+			userid = toId(parts[0].trim());
 			targetUser = Users.getExact(userid);
 			userBadges = Db.userBadges.get(userid);
-			selectedBadge = parts[2].trim();
-			if (!Db.badgeData.has(selectedBadge)) return this.errorReply("This badge does not exist, please check /badges list");
+			selectedBadge = parts[1].trim();
+			if (!Db.badgeData.has(selectedBadge)) return this.errorReply("This badge does not exist, please check /badges list.");
 			if (!Db.userBadges.has(userid)) userBadges = [];
 			userBadges = userBadges.filter(b => b !== selectedBadge);
 			userBadges.push(selectedBadge);
 			Db.userBadges.set(userid, userBadges);
-			if (Users.get(targetUser)) Users.get(userid).popup('|modal||html|You have received a badge from ' + WL.nameColor(toId(user), true) + ': <img src="' + Db.badgeData.get(selectedBadge)[1] + '" width="16" height="16"> (' + selectedBadge + ')');
-			this.logModCommand(user.name + " gave the badge '" + selectedBadge + "' badge to " + userid + ".");
-			this.sendReply("The '" + selectedBadge + "' badge was given to '" + userid + "'.");
-			break;
-		case 'create':
+			if (Users.get(targetUser)) Users.get(userid).popup(`|modal||html|You have received a badge from ${WL.nameColor(user.name, true)}: <img src="${Db.badgeData.get(selectedBadge)[1]}" width="16" height="16"> (${selectedBadge})`);
+			this.modlog(`BADGES`, targetUser, `gave badge`);
+			this.sendReply(`The "${selectedBadge}" badge was given to ${parts[0]}.`);
+		},
+
+		create: function (target, room, user) {
 			if (!this.can('ban')) return false;
-			if (parts.length !== 4) return this.errorReply("Correct command: `/badges create, [badge name], [description], [image]`.");
-			let badgeName = Chat.escapeHTML(parts[1].trim());
-			let description = Chat.escapeHTML(parts[2].trim());
-			let img = parts[3].trim();
+			let parts = target.split(',');
```

Comment:
`target.split().map(p => p.trim());` and remove all of the `trim()` from the following `parts[/* index */]`s.